### PR TITLE
Move CS resources extra resources to the CSV annotations

### PR DESCRIPTION
**Pull request by DanielXLee**
Test done with csv

### CODE REVIEW

File: pkg/bootstrap/init.go [hunk 1]
```diff
@@ -251,51 +200,33 @@ func yamlToObject(yamlContent []byte) (*unstructured.Unstructured, error) {
...
+					return err
+				}
+			}
 		}
 	}
-
-	return nil
-}
-
-func deleteExistingODLM(client client.Client) error {
```
Comment (horis233) [inline pkg/bootstrap/init.go:281]: Should we delete this section?

File: pkg/bootstrap/init.go [hunk 2]
```diff
@@ -41,86 +40,36 @@ import (
...
-		return err
+		return nil, fmt.Errorf("could not find the operator namespace: %v", err)
 	}
-	operandRegistry, err := ioutil.ReadFile(filepath.Join(resourcesDir, "cs-operandregistry.yaml"))
+	err = reader.Get(context.TODO(), types.NamespacedName{Name: deployName, Namespace: deployNs}, deploy)
 	if err != nil {
-		return err
-	}
-
-	timeout := time.After(300 * time.Second)
```
Comment (horis233) [inline pkg/bootstrap/init.go:99]: I think we still need to wait for the crd to be ready then deploy config and registry instances.

**Comment by horis233** [comment]
@DanielXLee I test the PR by copying the CSV and deploying the operator RBAC. The resources can be deployed. 
There are probably two cases that need to be added.
- Waiting for crds being ready
- Remove the 3.3 ODLM subscription and CSV.

File: pkg/bootstrap/init.go [hunk 3]
```diff
@@ -251,51 +200,33 @@ func yamlToObject(yamlContent []byte) (*unstructured.Unstructured, error) {
...
-	for _, file := range yamlFiles {
-		yamlContent, err := ioutil.ReadFile(filepath.Join(dir, file))
-		if err != nil {
-			return err
-		}
-		if err := createOrUpdateFromYaml(yamlContent, client, reader); err != nil {
-			return err
+	annotations := deploy.Spec.Template.GetAnnotations()
+	for _, res := range CsResources {
+		if res == "extraResources" {
```
Comment (chenzhiwei) [inline pkg/bootstrap/init.go:210]: You can separate this if into a function to reduce the nested levels and make the code more readable.

File: pkg/bootstrap/init.go (hunk 1 above)
Comment (chenzhiwei) [reply pkg/bootstrap/init.go:281]: @DanielXLee We still need to keep this part as we may release 3.4.1 to stable-v1 channel.

File: pkg/bootstrap/init.go [hunk 4]
```diff
@@ -251,51 +200,33 @@ func yamlToObject(yamlContent []byte) (*unstructured.Unstructured, error) {
 	return obj, nil
 }
 
-func createOrUpdateResourcesFromDir(dir string, client client.Client, reader client.Reader) error {
-	files, err := ioutil.ReadDir(dir)
+func createOrUpdateResourcesFromAnnotation(client client.Client, reader client.Reader) error {
```
Comment (chenzhiwei) [inline pkg/bootstrap/init.go:203]: I'd like to separate the CS resources and extra resources into two functions as the behavior before.

And you need to add wait for on the CS resources creation.

File: pkg/bootstrap/init.go (hunk 1 above)
Comment (DanielXLee) [reply pkg/bootstrap/init.go:281]: this func not reference, so I delete it.

Comment (horis233) [reply pkg/bootstrap/init.go:281]: It was referred in https://github.com/IBM/ibm-common-service-operator/blob/master/pkg/bootstrap/init.go#L67

Comment (DanielXLee) [reply pkg/bootstrap/init.go:281]: got it, back

File: pkg/bootstrap/init.go (hunk 3 above)
Comment (DanielXLee) [reply pkg/bootstrap/init.go]: Splited

File: pkg/bootstrap/init.go (hunk 4 above)
Comment (DanielXLee) [reply pkg/bootstrap/init.go]: done

File: pkg/bootstrap/init.go (hunk 2 above)
Comment (DanielXLee) [reply pkg/bootstrap/init.go:99]: done

File: pkg/bootstrap/init.go [hunk 5]
```diff
@@ -22,105 +22,127 @@ import (
...
-			if errRegistry != nil {
-				klog.Error("create OperandRegistry error with: ", errRegistry)
+func createOrUpdateResources(annotations map[string]string, resNames []string, client client.Client, reader client.Reader) error {
+	for _, res := range resNames {
+		if r, ok := annotations[res]; ok {
+			klog.Infof("create resource: %s", res)
+			if err := createOrUpdateFromYaml([]byte(r), client, reader); err != nil {
+				return err
 			}
+		}
```
[the remark this answers is not in the document]
Comment (DanielXLee) [reply pkg/bootstrap/init.go:111]: done

**Comment by renshichuang** [comment]
@chenzhiwei What's the related GHE? What's the impact?

**Comment by horis233** [comment]
@DanielXLee Sorry. When I tried to resolve the conflict from the Github, I involved some spaces in the CSV description. Please review and remove these spaces.

**Comment by DanielXLee** [comment]
@horis233 conflict solved

File: pkg/bootstrap/init.go [hunk 6]
```diff
@@ -22,105 +22,129 @@ import (
...
+	deploy := &appsv1.Deployment{}
+	deployName, err := k8sutil.GetOperatorName()
+	if err != nil {
+		return nil, fmt.Errorf("could not find the operator name: %v", err)
+	}
+	deployNs, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		return nil, fmt.Errorf("could not find the operator namespace: %v", err)
+	}
+	err = reader.Get(context.TODO(), types.NamespacedName{Name: deployName, Namespace: deployNs}, deploy)
```
Comment (chenzhiwei) [inline pkg/bootstrap/init.go:126]: @DanielXLee Let's add a loop to periodically get the deploy 3 times, interval 1 minute.

Comment (DanielXLee) [reply pkg/bootstrap/init.go]: added @chenzhiwei

**Pull request merged**